rust/xaynet-analytics/src/database/analytics_event/data_model.rs

```diff
@@ -0,0 +1,72 @@
+use chrono::{DateTime, Utc};
+use isar_core::object::{data_type::DataType, object_builder::ObjectBuilder};
+use std::vec::IntoIter;
+
+use crate::database::common::{FieldProperty, IsarAdapter};
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum AnalyticsEventType {
+    AppEvent,
+    Error,
+    ScreenEnter,
+    UserAction,
+}
+
+impl AnalyticsEventType {
+    pub fn to_int(&self) -> i32 {
+        match self {
+            AnalyticsEventType::AppEvent => 0,
+            AnalyticsEventType::Error => 1,
+            AnalyticsEventType::ScreenEnter => 2,
+            AnalyticsEventType::UserAction => 3,
+        }
+    }
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct AnalyticsEvent {
+    pub name: String,
+    pub event_type: AnalyticsEventType,
+    pub timestamp: DateTime<Utc>,
+    pub screen_route: Option<String>,
+}
+
+impl AnalyticsEvent {
+    pub fn new<N: Into<String>, R: Into<Option<String>>>(
```

Comment:
just went with [this comment](https://github.com/xaynetwork/xaynet/pull/673#discussion_r558105278)  from @Robert-Steiner 

i'm fine either way, don't know enough to have my own opinion

Reply:
personally i think `Into<String>` has some advantages
but yeah i agree `Into<Option<String>>` could be too much
feel free to remove it 😊 
There is a trick to avoid code bloat https://www.possiblerust.com/pattern/non-generic-inner-functions 
but maybe this it too early to apply these optimisations

Reply:
this sprint `ScreenRoute` will become a struct so the `String` stuff will be removed anyway, so perhaps with this PR we can just leave it.

very nice post though, thanks

Reply:
There is also a macro that does that but I forgot the name. It's interesting that putting `inner` inside does not bloat the code but seems compiler specific.